src/main/java/io/axoniq/axonserver/connector/event/impl/DcbEventChannelImpl.java

```diff
@@ -0,0 +1,225 @@
+import io.axoniq.axonserver.grpc.event.dcb.GetTailResponse;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.SourceEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsRequest;
+import io.axoniq.axonserver.grpc.event.dcb.StreamEventsResponse;
+import io.axoniq.axonserver.grpc.event.dcb.TaggedEvent;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * {@link DcbEventChannel} implementation, serving as the event connection between Axon Server and a client
+ * application.
+ *
+ * @author Milan Savic
+ * @since 2024.1.0
+ */
+public class DcbEventChannelImpl extends AbstractAxonServerChannel<Void> implements DcbEventChannel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DcbEventChannelImpl.class);
+
+    private static final int BUFFER_SIZE = 32;
+    private static final int REFILL_BATCH = 8;
+    private final DcbEventStoreGrpc.DcbEventStoreStub eventStore;
+    private final ClientIdentification clientIdentification;
+
+    /**
+     * Instantiate an {@link AbstractAxonServerChannel}.
+     *
+     * @param clientIdentification     the identification of the client
+     * @param executor                 a {@link ScheduledExecutorService} used to schedule reconnections
+     * @param axonServerManagedChannel the {@link AxonServerManagedChannel} used to connect to AxonServer
+     */
+    public DcbEventChannelImpl(ClientIdentification clientIdentification,
+                               ScheduledExecutorService executor,
+                               AxonServerManagedChannel axonServerManagedChannel) {
+        super(clientIdentification, executor, axonServerManagedChannel);
+        this.eventStore = DcbEventStoreGrpc.newStub(axonServerManagedChannel);
+        this.clientIdentification = clientIdentification;
+    }
+
+    @Override
+    public void connect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void reconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public void disconnect() {
+        // nothing to do here
+    }
+
+    @Override
+    public boolean isReady() {
+        return true;
+    }
+
+    @Override
+    public AppendEventsTransaction startTransaction() {
+        FutureStreamObserver<AppendEventsResponse> response = new FutureStreamObserver<>(null);
+        StreamObserver<AppendEventsRequest> clientStream = eventStore.append(response);
+        return new AppendEventsTransactionImpl(clientStream, response);
+    }
+
+    @Override
+    public ResultStream<StreamEventsResponse> stream(StreamEventsRequest request) {
+        AbstractBufferedStream<StreamEventsResponse, Empty> result =
```

Comment:
Shouldn't we collect all these buffers as internal state to the channel?
Looking at the `EventChannelImpl`, it keeps track of **any** before constructed during stream, which can then be closed on a `reconnect()` or `disconnect()`. I'd wager it would be valuable of the DCB channel mirrors that behavior for clients.